uvm/hier/usr/lib/python3/dist-packages/tests/global_functions.py

```diff
@@ -143,18 +141,251 @@ def verify_iperf_configuration(wan_ip):
     if iperf_serverReachable != 0:
         print("iperf Server is unreachable.")
         return False
-    # Check to see if some other test is using iperf for UDP testing
-    iperfRunning = remote_control.run_command("pidof iperf", host=iperf_server)
-    if iperfRunning == 0:
-        print("iperf is already running on server.")
-        return False
+    if ignore_running is False:
+        # Check to see if some other test is using iperf for UDP testing
+        iperfRunning = remote_control.run_command("pidof iperf", host=iperf_server)
+        if iperfRunning == 0:
+            print("iperf is already running on server.")
+            return False
     # Check that the client has iperf
     clientHasIperf = remote_control.run_command("test -x /usr/bin/iperf")
     if clientHasIperf != 0:
         print("iperf not installed on client.")
         return False
     return True
 
+# For parsing final iperf results:
+# [  3] 0.0000-10.0006 sec  9.59 GBytes  8.24 Gbits/sec
+# Also, transferred/rate can be non-float:
+# [  3] 0.0000-5.0005 sec   952 MBytes  1.60 Gbits/sec"
+iperf_results_re = re.compile('^\[\s+\d+\]\s+([0-9]*[.][0-9]*)\-([0-9]*[.][0-9]*)\s+sec\s+([0-9]*[.]*[0-9]*)\s+([^\s]+)\s+([0-9]*[.]*[0-9]*)\s+([^/]+)/(.+)')
+def get_iperf_results(duration=30):
+    """
+    Run iperf and return results
+    """
+    result = 0
+    start_time = None
+    end_time = None
+    transferred_amount = None
+    transferred_size = None
+    rate_amount = None
+    rate_size = None
+    rate_interval = None
+
+    iperf_results = remote_control.run_command(build_iperf_command(duration=duration), stdout=True)
+
+    last_line = iperf_results.splitlines()[-1]
+    matches = iperf_results_re.match(last_line)
+    transferred_value = 0
+    rate_value = 0
+    if matches is not None:
+        # If we don't get here either:
+        # - Something fundamentally wrong went wrong with the iperf command
+        # - Our regex failed
+        start_time = matches.group(1)
```

Comment:
Some variables are assigned but not used, are they necessary?

Reply:
Good point.  I should include those values as a "duration" field.